src/track/keys.cpp

```diff
@@ -29,11 +29,6 @@ void Keys::setSubVersion(const QString& subVersion) {
     m_subVersion = subVersion;
 }
 
-bool Keys::isValid() const {
-    return m_keyMap.global_key() != mixxx::track::io::key::INVALID ||
-            m_keyMap.global_key_text().length() > 0;
-}
```

Comment:
what was the motivation for removing this?

Reply:
This is function has tricked me, because It hides implementation details. In all cases one of these checks is sufficient.  
I think it is better to explicit look for global_key or global_key_text in the demanding code.

Reply:
Mhmm, but won't that cause inconsistency problem then when the `global_key` is valid but the `global_key_text` is empty? Or is that a state that is to be expected?

Reply:
The other way round can happen. There might be a key text that can not be recognized by Mixxx. Than `global_key` invalid but this function considers the whole object as valid, even if there is 'bla bla bla' stored as text, which is clearly invalid for a human reader. Because of this ambiguities it is better to check explicit the single members as desired in the calling code.